.github/workflows/checks.yml

```diff
@@ -61,12 +61,19 @@ jobs:
       - name: Setup asdf
         uses: asdf-vm/actions/setup@v2
 
-      - name: Install sncast 0.6.0
+      - name: Install sncast
         run: |
           asdf plugin add starknet-foundry
           asdf install starknet-foundry 0.6.0
           asdf global starknet-foundry 0.6.0
 
+      - name: Install scarb
+        run: |
+          asdf plugin add scarb
+          asdf install scarb 0.4.0
+          asdf install scarb 0.7.0
+          asdf install scarb 2.6.0
+
```

Comment:
you can probably just use scarb and snfoundry actions to set things up. They support using .tool-versions files so we will avoid hardcoding versions in multiple places. 

https://github.com/marketplace/actions/setup-scarb
https://github.com/marketplace/actions/setup-starknet-foundry

btw what's the reason to use this very old sncast version?

Reply:
> you can probably just use scarb and snfoundry actions to set things up.

Does `setup-scarb` action support multiple versions? Originally, the [idea](https://github.com/software-mansion/starknet-jvm/pull/359#discussion_r1413655125) was to support multiple `scarb` versions to complile contracts written in different Cairo versions. I'm not sure that will be possible with scarb action.

> They support using .tool-versions files so we will avoid hardcoding versions in multiple places.

From what I understand from README `.tool-versions` will only be read once, hence option to pass `.tool-versions` path, but maybe I'm wrong.

> btw what's the reason to use this very old sncast version?

We had to keep it for a while because of the declare/path related bugs, didn't get around to fix it afterwards, but I think we can open issue for that, esp. considering it would simplify setting up tx v3 tests

Reply:
We use multiple `.tool-versions` files and `scarb` action allows to specify only one, so IIUC it isn't suitable for our case.

Reply:
as far as I remember, github actions are basically docker layers - so it should be possible to stack multiple setup actions with different versions, but let's verify.

> We had to keep it for a while because of the declare/path related bugs, didn't get around to fix it afterwards, but I think we can open issue for that, esp. considering it would simplify setting up tx v3 tests

I think we already discussed this a few times and there were no bugs, but let's also try and verify.

Reply:
if using scarb action is not possible, let's just use asdf action